lib/transport/transport-factory-socket.c

```diff
@@ -24,13 +24,14 @@
 
 #include "transport/transport-factory-socket.h"
 #include "transport/transport-socket.h"
+#include "transport/transport-udp-socket.h"
 
 DEFINE_TRANSPORT_FACTORY_ID_FUN("socket", transport_factory_socket_id);
 
 static LogTransport *
 _construct_transport_dgram(const TransportFactory *s, gint fd)
 {
-  return log_transport_dgram_socket_new(fd);
```

Comment:
I was thinking the new implementation serves as a drop-in replacement for the original implementation. Do we need to old code keep around, or it can be eliminated?

Reply:
This is a good point. as I see nothing really uses the base class anymore. Maybe some functionality could be merged by changing internal signatures.

Do you think I can delay that work once we merge this? I mean this is already pretty large and I guess incorporating that change into this PR would not make it smaller.

Reply:
Sure, this can be done later.